### ISSUE TITLE
sesame tokens seem to be missing a bunch of entropy on my Django installation (first characters are all 'AAAAAA'

### ISSUE THREAD

**Issue by joshburnett** [aaugustin/django-sesame#80]
I love this library! Solves the 'magic link' use case very nicely. Thanks for the hard work on it.

When I do a `get_token(user)` I get something like `AAAAAAAAABuD8yi2jSL1pyfz` back for all of my users. I'm running Django 4.0 with a user model that's subclassed from AbstractUser, if that's relevant. Happy to provide other info as needed to help debug.

**Comment by adamchainz** [comment]
This is expected behaviour - sesame uses the `struct` library to encode your user primary key: https://github.com/aaugustin/django-sesame/blob/main/src/sesame/packers.py . It then base64 encodes this: https://github.com/aaugustin/django-sesame/blob/7edd0a4ee4fc3f63c56077cbace4dae6daee3c1d/src/sesame/tokens_v2.py#L124-L139 .

For `BigAutoField`, it will encodes 8 bytes, for example for user ID 1 you'll see `AAAAAAAAAAE` (the trailing `=` is not required):

```pycon
>>> import struct
>>> struct.pack('!q', 1)
b'\x00\x00\x00\x00\x00\x00\x00\x01'
>>> import base64
>>> base64.b64encode(struct.pack('!q', 1))
b'AAAAAAAAAAE='
```

Perhaps the docs examples could be updated to show this. Would you like to make a PR?

**Comment by joshburnett** [comment]
Ahh, I see. The docs do briefly mention this in the 'Tokens security' section but don't give an example, so I just rolled right over it. I didn't think about it until I noticed that the tokens that I'm generating look very different than the example ones shown.

I'll think about pulling together a PR to put just a little more detail there. I'll probably also investigate how hard it is to write a custom packer, though I'm not sure it's really worth it for my particular low-stakes application.

Thanks again for creating this library, and for clearly already putting a bunch of work into the docs. Overall they are really quite good!
